platform/core/src/classes/ImageSet.ts

```diff
@@ -79,63 +75,96 @@ class ImageSet {
     return this.images[index];
   }
 
+  /**
+   * Default image sorting. Sorts by the following (in order of priority)
+   * 1. Image position (if ImagePositionPatient and ImageOrientationPatient are defined)
+   * 2. InstanceNumber
+   * Note: Images are sorted in-place and a reference to the sorted image array is returned.
+   *
+   * @returns images - reference to images after sorting
+   */
+  sort(sortingCallback: (a: Image, b: Image) => number): Image[] {
+    try {
+      return this.sortByImagePositionPatient();
+    } catch (error) {
+      return this.sortBy(sortingCallback);
+    }
+  }
+
+  /**
+   * Sort using the provided callback function.
+   * Note: Images are sorted in-place and a reference to the sorted image array is returned.
+   *
+   * @param sortingCallback - sorting function
+   * @returns images - reference to images after sorting
+   */
   sortBy(sortingCallback: (a: Image, b: Image) => number): Image[] {
     return this.images.sort(sortingCallback);
   }
 
-  sortByImagePositionPatient(): void {
-    const images = this.images;
-    const referenceImagePositionPatient = _getImagePositionPatient(images[0]);
-
-    const refIppVec = new Vector3(
-      referenceImagePositionPatient[0],
-      referenceImagePositionPatient[1],
-      referenceImagePositionPatient[2]
-    );
-
-    const ImageOrientationPatient = _getImageOrientationPatient(images[0]);
-
-    const scanAxisNormal = new Vector3(
-      ImageOrientationPatient[0],
-      ImageOrientationPatient[1],
-      ImageOrientationPatient[2]
-    ).cross(
-      new Vector3(
-        ImageOrientationPatient[3],
-        ImageOrientationPatient[4],
-        ImageOrientationPatient[5]
-      )
-    );
-
-    const distanceImagePairs = images.map(function (image: Image) {
-      const ippVec = new Vector3(..._getImagePositionPatient(image));
-      const positionVector = refIppVec.clone().sub(ippVec);
-      const distance = positionVector.dot(scanAxisNormal);
-
-      return {
-        distance,
-        image,
-      };
+  /**
+   * Sort by InstanceNumber
+   * Note: Images are sorted in-place and a reference to the sorted image array is returned.
+   *
+   * @returns images - reference to images after sorting
+   */
+  sortByInstanceNumber(): Image[] {
+    return this.sortBy((a, b) => {
+      // Sort by InstanceNumber (0020,0013)
+      // @ts-ignore
+      return (parseInt(a.InstanceNumber) || 0) - (parseInt(b.InstanceNumber) || 0);
     });
+  }
 
-    distanceImagePairs.sort(function (a, b) {
-      return b.distance - a.distance;
-    });
+  /**
+   * Sort by image position, calculated using ImageOrientationPatient and ImagePositionPatient
+   * Note: Images are sorted in-place and a reference to the sorted image array is returned.
+   *
+   * @returns images - reference to images after sorting
+   */
+  sortByImagePositionPatient(): Image[] {
```

Comment:
With the remaining fixes for making this a comparator, the createImagePositionPatientCompare should get moved into sortStudy so that it can be re-used consistently.